pkg/ui/src/views/shared/components/sortedtable/index.tsx

```diff
@@ -121,15 +131,19 @@ export class SortedTable<T> extends React.Component<SortedTableProps<T>, {}> {
   );
 
   render() {
-    const { data, sortSetting, onChangeSortSetting } = this.props;
+    const { data, sortSetting, onChangeSortSetting, loading } = this.props;
+    if (!_.isNil(loading) && loading) {
+      return <div className="sorted-table-section" style={spinnerImage} />;
+    }
     if (data) {
       return <SortableTable count={data.length}
-                            sortSetting={sortSetting}
-                            onChangeSortSetting={onChangeSortSetting}
-                            columns={this.columns(this.props)}
-                            rowClass={this.rowClass(this.props)}
-                            className={this.props.className} />;
+        sortSetting={sortSetting}
+        onChangeSortSetting={onChangeSortSetting}
+        columns={this.columns(this.props)}
+        rowClass={this.rowClass(this.props)}
+        className={this.props.className}
+        loading={this.props.loading} />;
```

Comment:
You're duplicating work here; both sortedTable and sortableTable can display a 'spinner' gif section, it should either be one or the other.

But again, as stated above, it should be a separate component.